ProxyAgent.Test/ProxyTest.cs

```diff
@@ -128,30 +130,34 @@ public void ItDoesntProxySomeHeaders()
             // Act
             this.target.ProcessAsync("https://" + request.Host, request, response).Wait(TestTimeout);
 
-            // Assert - requests headers are blocked/passed as expected
+            // Assert - these request headers are allowed
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => r.Headers.Contains("Content-Type"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => r.Headers.Contains("Cookie"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => r.Headers.Contains("X-ActivityId"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => r.Headers.Contains("X-CorrelationId"))), Times.Once);
+
+            // Assert - these request headers are blocked
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Connection"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Content-Length"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Keep-Alive"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Host"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Upgrade"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Upgrade-Insecure-Requests"))), Times.Once);
```

Comment:
#supernit Unsecure, not Insecure

Reply:
can't fix the web :-) https://www.w3.org/TR/upgrade-insecure-requests